src/objects/Mesh.js

```diff
@@ -107,6 +111,21 @@ THREE.Mesh.prototype.raycast = ( function () {
 
 		var a, b, c;
 
+		var textureIntersection = function ( pIntersection, p1, p2, p3, uv1, uv2, uv3 ) {
+
+			var bary = THREE.Triangle.barycoordFromPoint( pIntersection, p1, p2, p3 );
+
```

Comment:
`THREE.Triangle.barycoordFromPoint()` instantiates a new `Vector3()` with each call. To prevent that, you need to pass in an `optionalTarget`.

You can use the prototype method with the same name, instead, if you want.